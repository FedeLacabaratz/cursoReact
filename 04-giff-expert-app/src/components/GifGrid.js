import React, { useState, useEffect } from 'react';
import GifGridItem from './GifGridItem';
import getGifs from '../helpers/getGifs';

const GifGrid = ({ category }) => {

    const [images, setImages] = useState([])

    useEffect(() => {
        getGifs( category )
            .then(setImages);
    }, [category])

    return (
        <>
            <h3>{category}</h3>
            <ol className="card-grid">
                {images.map((img) => (
                    <GifGridItem
                        key={img.id}
                        {...img}
                    />
                ))}
            </ol>
        </>
    );
}

export default GifGrid;