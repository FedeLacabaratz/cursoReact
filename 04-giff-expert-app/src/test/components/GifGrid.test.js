import React from 'react';
import '@testing-library/jest-dom';
import GifGrid from '../../components/GifGrid';
import { shallow } from 'enzyme';
import useFetchGifs from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('pruebas en <GifGrid />', () => {
     
    let category = 'categoria';
    
    beforeEach(() => {
        jest.clearAllMocks();
    });
    
    test('debe de imprimir el componente correctamente', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imÃ¡genes con nuestro useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/any/thing.jpg',
            title: 'Anything',
        },
        {
            id: '123',
            url: 'https://localhost/any/thing.jpg',
            title: 'Anything',
        }]
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });

});
