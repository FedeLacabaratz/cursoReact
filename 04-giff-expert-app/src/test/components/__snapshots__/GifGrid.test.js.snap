// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifGrid /> debe de imprimir el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    categoria
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <ol
    className="card-grid"
  />
</Fragment>
`;

exports[`pruebas en <GifGrid /> debe de mostrar items cuando se cargan im√°genes con nuestro useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    categoria
  </h3>
  <ol
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Anything"
      url="https://localhost/any/thing.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Anything"
      url="https://localhost/any/thing.jpg"
    />
  </ol>
</Fragment>
`;
